pkg/cmd/cli/cmd/newapp.go

```diff
@@ -532,6 +532,10 @@ func CompleteAppConfig(config *newcmd.AppConfig, f *clientcmd.Factory, c *cobra.
 	if config.BinaryBuild && config.Strategy == generate.StrategyPipeline {
 		return kcmdutil.UsageError(c, "specifying binary builds and the pipeline strategy at the same time is not allowed.")
 	}
+
+	if len(config.BuildArgs) > 0 && config.Strategy != generate.StrategyDocker {
+		return kcmdutil.UsageError(c, "build arguments can only be used with Docker builds.")
+	}
```

Comment:
You're right, I didn't catch generate.StrategyUnspecified. I've added that to the check. Do we want to error out if the Build doesn't use the BuildArgs? Currently, the BuildArgs will not be used if no ComponentReferences have a DockerStrategy.

However, since we'll likely only have one SourceRepository while using `new-build`, we could throw an error if the Strategy for that is not DockerStrategy.